### DISCUSSION TITLE
Can't open a file.

### DISCUSSION THREAD

**Discussion by Zeyadas** [koreader/koreader#7145] [Q&A] [answered]
Hi. Didn't want to use the issue section for this. 

Using an android tablet and the latest version of koreader. 

I couldn't open this file (attached below and added the.zip) on koreader even though it opened (but poorly displayed) on bookari premium 
![Screenshot_2021-01-15-10-45-01](https://user-images.githubusercontent.com/70109540/104696070-bc04d200-571e-11eb-91e2-0b0f2e2dfad9.png)
The file:
[t.epub.zip](https://github.com/koreader/koreader/files/5819064/t.epub.zip)

**Comment by Frenzie** [comment, marked as answer]
I don't think I've ever seen so many errors in epubcheck, but on further inspection that's mainly because of the `content.opf`.

The document *repeats* the same pages dozens (or even hundreds?) of times! So what would be a 300 page document is stretched out into a 10 000 page document.

It loads, but depending on the speed of your Android device you might have to wait quite a while. Also Android will incorrectly display a "not responding" notification in that case. Or well, not incorrectly exactly, but in any case it's not responding because it's too busy loading the document.

tl;dr That EPUB is very, very broken.

**Reply by Zeyadas** [reply]
I found that (file/book) in a library that has over 29K books, and I downloaded it ALL. so I wasted my time and my poor internet connection made it worse.
So many thanks guys for responding and I liked the (tl;dr), that was helpful.
I wish I can help you out somehow, I'm so grateful.

**Reply by Frenzie** [reply]
I've attached a version I quickly fixed up in Sigil (remove .zip from the name, same as the one you uploaded).
[t.epub.zip](https://github.com/koreader/koreader/files/5823266/t.epub.zip)

**Reply by Zeyadas** [reply]
Worked perfectly, thank you. 
WHAT IS (sigil), how can I fix the rest of the files I have, and I don't have a PC. Just two android devices.

**Reply by Frenzie** [reply]
Sigil is https://sigil-ebook.com/

But I don't think it can help with this. I removed the thousands of lines of duplicates from content.opf manually.

**Reply by Zeyadas** [reply]
Oh, thanks.
I thought it'd be an easy job to do by myself for thousands of files I have.
